lib/system/generic/xlnx/sys.c

```diff
@@ -72,12 +72,63 @@ void metal_weak metal_generic_default_poll(void)
 	metal_asm volatile("wfi");
 }
 
+/*
+ * VERSAL_NET is used since XMpu_Config structure is
+ * different for r52(versal net) and r5(zynqmp) to avoid build failure
+ */
+#ifdef VERSAL_NET
+void *metal_machine_io_mem_map_versal_net(void *va, metal_phys_addr_t pa,
+					  size_t size, unsigned int flags)
+{
+	void *__attribute__((unused)) physaddr;
+	u32 req_end_addr = pa + size;
+	XMpu_Config mpu_config;
+	u32 req_addr = pa;
+	u32 mmap_req = 1;
+	u32 base_end_addr;
+	u32 cnt;
+
+	/* Get the MPU Config enties */
+	Xil_GetMPUConfig(mpu_config);
+
+	for (cnt = 0; (cnt < MAX_POSSIBLE_MPU_REGS) &&
+	     (mpu_config[cnt].flags & XMPU_VALID_REGION); cnt++) {
```

Comment:
This test doesn't seem quite right. If the MPU registers are sparse, for example 0,1 and 3 have valid regions, but 2 does not, then 3 will be skipped in this loop. It's been quite awhile since I looked at the MPU specs, but I think the hardware allows this.

Maybe this test should be moved inside the loop, with an if() continue statement?

Reply:
Thanks @edmooring It's done.